scalaparse/shared/src/main/scala/scalaparse/syntax/Literals.scala

```diff
@@ -32,6 +32,7 @@ trait Literals { l =>
     val ConsumeComments = P( (Basic.WSChars.? ~ Literals.Comment ~ Basic.WSChars.? ~ Basic.Newline).rep )
     P( NoCut( WS ~ Basic.Newline.? ~ ConsumeComments ~ NotNewline) )
   }
+  val TrailingComma: P0 = ("," ~ WS.rep ~ Basic.Newline).?
```

Comment:
This `.rep` is unnecessary, since the `WS` internally already loops 0-or-more times, and the fact that `WS` can succeed when consuming 0 characters means the surrounding `.rep` can succeed infinitely without making forward progress. I have removed it, and all your new tests cases still seem to pass, so that's probably fine